packages/blade/src/components/Tag/_decisions/decisions.md

```diff
@@ -0,0 +1,198 @@
+
+  /**
+   * Callback when close icon on Tag is clicked
+   *
+   * Gives a `value` argument which has same text as `children`
+   */
+  onDismiss?: ({ value }: { value: TagProps['children'] }) => void;
+
+  /**
+   * Text that renders inside Tag
+   */
+  children: ReactText;
+
+  /**
+   * Disable tag
+   */
+  isDisabled?: boolean;
+};
+```
+
+_Didn't consider any alternate API. Kept `size` and `children` similar to `<Badge />` and `onDismiss` similar to `<Alert />` and `<Dropdown />`_
+
+## Integration with Inputs
+
+<img width="358" alt="image" src="https://github.com/razorpay/blade/assets/30949385/71cd236a-93e7-46f0-9da6-12d519b958d4">
+
+Tags can be added inside other input components such as -
+
+- TextArea
+- SelectInput (in Dropdown)
+- TextInput
+- AutoComplete
+
+<details>
+<summary>Implementation Details</summary>
+
+This will require some refactor in BaseInput to add a slot before the actual Input element.
+
+E.g. This is AutoComplete from Primer where the tags go into a slot and input gets pushed forward. We will be implementing something similar in our BaseInput.
+
+<img width="419" alt="image" src="https://github.com/razorpay/blade/assets/30949385/be01cdfb-1171-4970-b381-76e4c44d1854">
+
+</details>
+
+### Integration with SelectInput and AutoComplete
+
+SelectInput and AutoComplete are not expected to have any changes in consumer API as rendering of tags will be handled internally based on the selected value.
+
+### Integration with TextInput and TextArea
+
+> **Note**
+>
+> This implementation is out of scope for initial release of Tag.
+>
+> Have added considered approaches below which we can refer back and confirm the feasibility while implementation. (Feel free to review and suggest any changes here though. We can consider them while imeplementation in future)
+
+On consumer end, the APIs would look like -
+
+#### New `tagsSlot` prop on `TextInput` and `TextArea`
+
+- Can only be used in Controlled Input
+  - Why?
+    - This is comparitively a rare usecase so don't see any need of having another `hasTags` props that handles adding tags internally.
+
+```jsx
+<TextInput tagsSlot={} value="" onChange={} />
+```
+
+##### Alternate Prop Name
+
+- `leading` (felt confusing because it is leading to the invisible input tag and not the actual visible TextInput)
+- `valuePrefix` (Similar to `titleSuffix` from BaseHeader)
+
+<details>
+<summary>Full Controlled Example</summary>
+
+```jsx
+function App() {
+  const [inputValue, setInputValue] = React.useState('');
+  const [tags, setTags] = React.useState([]);
+
+  const addTag = () => {
+    // Add input value to tags and clear the input value
+    setTags([...tags, inputValue]);
+    setInputValue('');
+  }
+
+  const removeTag = (tagName) => {
+    setTags(tags.filter(tagNameValue) => tagNameValue !== tagName);
+  }
+
+  return (
+    <TextInput
+      tagsSlot={tags.map((tagName, index) => (
+        <Tag onDismiss={({ value }) => removeTag(value)}>{tagName}</Tag>
```

Comment:
Should we also think about some `id` based prop? Eg. for handling duplicate tags (having same names)

Reply:
We'll also need `key` prop here

Reply:
`key` is automatically supported in react and typescript. It doesn't seem to throw any error.

By `id` you meant something like `value` as you mentioned in the comment below 🤔? Makes sense. Should it be optional?

Reply:
Gave this a thought and we might not need this because if someone has unique tags, they can also have unique ids on their end and pass that id instead of value returned from `onDismiss`.


E.g.
```jsx
const tags = [{ id: 'red-a', name: 'Red' }, { id: 'red-b', name: 'Red' }]

tags.map(([tagId, tagName], index) => (
     <Tag onDismiss={() => removeTag(tagId)}>{tagName}</Tag>
)
```

Initially I had thought its fine to not even have `value` argument from onDismiss but just added it incase someone wants to do this -`onDismiss={dismissHandler}` (then they won't have to pass the tagName separately)

Should we name argument to something like this? - 
1. tagName
2. name

Reply:
I think `value` should be good enough, `key` / `id` based things can be skipped 👍

Reply:
Continued this discussion on slack. Summary was we can go without value prop because consumers can handle this on their own with something like this -
```jsx
<Tag onDismiss={() => doSomething('sm-1')}>Small</Tag>
<Tag onDismiss={() => doSomething('sm-2')}>Small</Tag>
```

This is also the approach that other design systems have
- [Carbon passes no argument from onDismiss param](https://react.carbondesignsystem.com/?path=/story/components-tag--playground&args=filter:!true)
- [Atlassian passes text prop which is the same prop that they use for display value as well](https://atlassian.design/components/tag/code)

We decided we can asses value prop later.